source/sigv4.c

```diff
@@ -1178,6 +1189,14 @@ static SigV4Status_t generateCredentialScope( const SigV4Parameters_t * pSigV4Pa
         return URI_DOUBLE_ENCODED_EQUALS_CHAR_SIZE;
     }
 
+/*-----------------------------------------------------------*/
+
+    static bool isAllowedChar( char c,
+                               bool encodeSlash )
+    {
+        return( isalnum( c ) || ( c == '-' ) || ( c == '_' ) || ( c == '.' ) || ( c == '~' ) || ( ( c == '/' ) && !encodeSlash ) );
```

Comment:
```suggestion
        return( isalnum( c ) || ( c == '-' ) || ( c == '_' ) || ( c == '.' ) || ( c == '~' ) || ( ( c == '/' ) && ( encodeSlash == false ) ) );
```

Reply:
Since I'm touching this line in #33 as well, I think I'll address it then. This line also uses `isalnum`, which we might have to reimplement to comply with MISRA directive 4.6, so this might be converted into an `if-else if-else` block anyway